src/tests/ftest/nvme/nvme_pool_extend.py

```diff
@@ -0,0 +1,242 @@
+        """[summary]
+        Args:
+            action (str): Start the IOR thread with Read or
+                          Write
+            oclass (str): IOR object class
+            API (str): IOR API
+            test (list): IOR test sequence
+            flags (str): IOR flags
+        """
+        if action == "Write":
+            flags = self.ior_w_flags
+        else:
+            flags = self.ior_r_flags
+
+        # Add a thread for these IOR arguments
+        process = threading.Thread(target=self.ior_thread,
+                                   kwargs={"pool": self.pool,
+                                           "oclass": oclass,
+                                           "api": api,
+                                           "test": test,
+                                           "flags": flags,
+                                           "results":
+                                           self.out_queue})
+        # Launch the IOR thread
+        process.start()
+        # Wait for the thread to finish
+        process.join()
+
+    def run_nvme_pool_extend(self, num_pool):
+        """Run Pool Extend
+            Args:
+             int : total pools to create for testing purposes.
+        """
+        # Create a pool
+        pool = {}
+        total_servers = len(self.hostlist_servers) * 2
+
+        # Extend one of the ranks (or server)
+        # rank index starts from zero
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context, dmg_command=self.dmg_command)
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            for oclass, api, test in product(self.ior_dfs_oclass,
+                                             self.ior_apis,
+                                             self.ior_test_sequence):
+                self.run_ior_thread("Write", oclass, api, test)
+
+                scm_size = self.pool.scm_size
+                nvme_size = self.pool.nvme_size
+                self.pool.display_pool_daos_space("Pool space: Beginning")
+                pver_begin = self.get_pool_version()
+
+        # Start the additional servers and extend the pool
+        self.log.info("Extra Servers = %s", self.extra_servers)
+        self.start_additional_servers(self.extra_servers)
+        # Give sometime for the additional server to come up.
+        time.sleep(25)
+        self.log.info("Pool Version at the beginning %s", pver_begin)
+        output = self.dmg_command.pool_extend(self.pool.uuid,
```

Comment:
Sorry I forgot in my previous review, But we need to extend all the pool so it should be in for loop for for val in range(0, num_pool) , right?

Reply:
Hmm.. We had it in the loop. But, we cannot start additional servers for every loop.  I think we can use this rank for each pool created is it not? We start a server with one rank and we can extend the multiple pools on this rank. Is it not? Presently, we need to think in detail for multiple pools. Should work only for single pool now.